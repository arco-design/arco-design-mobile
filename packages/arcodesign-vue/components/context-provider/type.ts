export interface ContextInstallOptions {
    /**
     * 组件名称前缀
     * @en Component prefix
     * @default "Arco"
     */
    componentPrefix?: string;
}

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
}
