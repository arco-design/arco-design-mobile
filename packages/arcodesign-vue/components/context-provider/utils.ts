import { InjectionKey, inject } from 'vue';
import { ContextInstallOptions, GlobalContextParams } from './type';

export const defaultInstallOptions: Required<ContextInstallOptions> = {
    componentPrefix: 'Arco',
};

export const defaultContext: Required<GlobalContextParams> = {
    prefixCls: 'arco',
};

export const configProviderInjectionKey: InjectionKey<GlobalContextParams> =
    Symbol('ArcoContextProvider');

export function getComponentPrefix(options?: ContextInstallOptions) {
    return options?.componentPrefix || defaultInstallOptions.componentPrefix;
}

export function getPrefixCls(componentName?: string) {
    const contextProvider = inject(configProviderInjectionKey, defaultContext);
    const prefix = contextProvider.prefixCls;
    if (componentName) {
        return `${prefix}-${componentName}`;
    }
    return prefix;
}
