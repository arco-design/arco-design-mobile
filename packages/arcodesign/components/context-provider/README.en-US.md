### Others

# ContextProvider 

Global data control component, used to replace global data.

======

> Props

|Property|Description|Type|DefaultValue|
|----------|-------------|------|------|
|prefixCls|Component classname prefix|string|"arco"|
|system|Manually control the current system, and the incoming value will be used directly after being passed in\. It is applicable when the initial value of the system needs to be specified in the ssr scenario\.|"" \| "pc" \| "android" \| "ios"|""|
|useDarkMode|Whether to use dark mode|boolean|false|
|isDarkMode|Whether it is in dark mode|boolean|false|
|theme|Theme variable\. The css variable will be replaced online after input\. The less variable needs to be set|Record\<string, string\>|-|
|locale|Internationalized language configuration|ILocale|-|
|useRtl|Whether to use rtl|boolean|false|

> ILocale

|Property|Description|Type|
|----------|-------------|------|
|locale|Language Type|string|
|LoadMore|-|\{ loadMoreText: string; loadingText: string; prepareText: string; noDataText: string; failLoadText: string; prepareScrollText: string; prepareClickText: string; \}|
|Picker|-|\{ okText: string; cancelText: string; \}|
|Tag|-|\{ addTag: string; \}|
|Dialog|-|\{ okText: string; cancelText: string; \}|
|SwipeLoad|-|\{ normalText: string; activeText: string; \}|
|PullRefresh|-|\{ loadingText: string; pullingText: string; finishText: string; loosingText: string; \}|
|DropdownMenu|-|\{ select: string; \}|
|Pagination|-|\{ previousPage: string; nextPage: string; \}|
|Image|-|\{ loadError: string; \}|
|ImagePicker|-|\{ loadError: string; \}|
|SearchBar|-|\{ placeholder: string; cancelBtn: string; \}|
|Stepper|-|\{ minusButtonName: string; addButtonName: string; \}|
|Keyboard|-|\{ confirm: string; \}|
|Form|-|\{ required: string; type: \{ email: string; url: string; string: string; number: string; array: string; object: string; boolean: string; \}; number: \{ min: string; max: string; equal: string; range: string; positive: string; negative: string; \}; string: \{ \.\.\.; \}; array: \{ \.\.\.; \}; object: \{ \.\.\.; \}; boolean: \{ \.\.\.; \}; \}|
