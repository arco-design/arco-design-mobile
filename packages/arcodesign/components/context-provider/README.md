### 其他

# 全局配置 ContextProvider

全局数据控制组件，用于替换全局数据。

======

> 属性/Props

|参数|描述|类型|默认值|
|----------|-------------|------|------|
|prefixCls|组件类名前缀|string|"arco"|
|system|手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用|"" \| "pc" \| "android" \| "ios"|""|
|useDarkMode|是否使用暗黑模式|boolean|false|
|isDarkMode|是否处于暗黑模式，指定后以指定的值为准|boolean|false|
|theme|主题变量，传入后将在线替换css变量，需设置less变量 @use\-css\-vars: 1|Record\<string, string\>|-|
|locale|国际化语言包配置|ILocale|-|
|useRtl|是否使用Rtl模式|boolean|false|

> ILocale

|参数|描述|类型|
|----------|-------------|------|
|locale|语言类型|string|
|LoadMore|-|\{ loadMoreText: string; loadingText: string; prepareText: string; noDataText: string; failLoadText: string; prepareScrollText: string; prepareClickText: string; \}|
|Picker|-|\{ okText: string; cancelText: string; \}|
|Tag|-|\{ addTag: string; \}|
|Dialog|-|\{ okText: string; cancelText: string; \}|
|SwipeLoad|-|\{ normalText: string; activeText: string; \}|
|PullRefresh|-|\{ loadingText: string; pullingText: string; finishText: string; loosingText: string; \}|
|DropdownMenu|-|\{ select: string; \}|
|Pagination|-|\{ previousPage: string; nextPage: string; \}|
|Image|-|\{ loadError: string; \}|
|ImagePicker|-|\{ loadError: string; \}|
|SearchBar|-|\{ placeholder: string; cancelBtn: string; \}|
|Stepper|-|\{ minusButtonName: string; addButtonName: string; \}|
|Keyboard|-|\{ confirm: string; \}|
|Form|-|\{ required: string; type: \{ email: string; url: string; string: string; number: string; array: string; object: string; boolean: string; \}; number: \{ min: string; max: string; equal: string; range: string; positive: string; negative: string; \}; string: \{ \.\.\.; \}; array: \{ \.\.\.; \}; object: \{ \.\.\.; \}; boolean: \{ \.\.\.; \}; \}|
