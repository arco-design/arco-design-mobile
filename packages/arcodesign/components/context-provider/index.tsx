import React, { createContext, useState, useEffect, useCallback } from 'react';
import { addCssRules, removeCssStyleDom, defaultLocale, ILocale } from '@arco-design/mobile-utils';

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
    /**
     * 手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用
     * @en Manually control the current system, and the incoming value will be used directly after being passed in. It is applicable when the initial value of the system needs to be specified in the ssr scenario.
     * @default ""
     */
    system?: 'pc' | 'android' | 'ios' | '';
    /**
     * 是否使用暗黑模式
     * @en Whether to use dark mode
     * @default false
     */
    useDarkMode?: boolean;
    /**
     * 是否处于暗黑模式，指定后以指定的值为准
     * @en Whether it is in dark mode
     * @default false
     */
    isDarkMode?: boolean;
    /**
     * 主题变量，传入后将在线替换css变量，需设置less变量 @use-css-vars: 1
     * @en Theme variable. The css variable will be replaced online after input. The less variable needs to be set @use-css-vars: 1
     */
    theme?: Record<string, string>;
    /**
     * 国际化语言包配置
     * @en Internationalized language configuration
     */
    locale?: ILocale;
    /**
     * 是否使用Rtl模式
     * @en Whether to use rtl
     * @default false
     */
    useRtl?: boolean;
}

export const defaultContext: GlobalContextParams = {
    prefixCls: 'arco',
    system: '',
    useDarkMode: false,
    isDarkMode: false,
    locale: defaultLocale,
    useRtl: false,
};

export const GlobalContext = createContext<GlobalContextParams>(defaultContext);

export interface ContextProviderProps extends GlobalContextParams {
    children: React.ReactNode;
}

export type WithGlobalContext<T> = T & { context?: GlobalContextParams };

/**
 * 全局数据控制组件，用于替换全局数据。
 * @en Global data control component, used to replace global data.
 * @type 其他
 * @type_en Others
 * @name 全局配置
 * @name_en ContextProvider
 */
export default function ContextProvider(props: ContextProviderProps) {
    const {
        children,
        useDarkMode,
        isDarkMode,
        theme,
        locale = defaultLocale,
        ...restProps
    } = props;
    const [isDarkModeState, setIsDarkModeState] = useState<boolean>();
    const changeDarkMode = useCallback(
        (res: MediaQueryListEvent) => setIsDarkModeState(res.matches),
        [],
    );

    useEffect(() => {
        const matchMedia = window.matchMedia('(prefers-color-scheme: dark)');
        if (useDarkMode) {
            const dark = matchMedia.matches;
            setIsDarkModeState(dark);
            if (typeof matchMedia.addEventListener === 'function') {
                matchMedia.addEventListener('change', changeDarkMode);
            } else if (typeof matchMedia.addListener === 'function') {
                matchMedia.addListener(changeDarkMode);
            }
        }
        return () => {
            if (useDarkMode) {
                if (typeof matchMedia.removeEventListener === 'function') {
                    matchMedia.removeEventListener('change', changeDarkMode);
                } else if (typeof matchMedia.removeListener === 'function') {
                    matchMedia.removeListener(changeDarkMode);
                }
            }
        };
    }, [useDarkMode]);

    useEffect(() => {
        if (!theme) {
            return;
        }
        addCssRules('arcoCustomTheme', theme);
        return () => {
            removeCssStyleDom('arcoCustomTheme');
        };
    }, [theme]);

    return (
        <GlobalContext.Provider
            value={{
                ...defaultContext,
                ...restProps,
                isDarkMode: isDarkMode === void 0 ? isDarkModeState : isDarkMode,
                useDarkMode,
                locale: locale || defaultLocale,
            }}
        >
            {children}
        </GlobalContext.Provider>
    );
}

export const ContextLayout = GlobalContext.Consumer;

export function CompWithGlobalContext<P extends JSX.IntrinsicAttributes>(
    Component: React.FunctionComponent<P>,
) {
    return function (props: WithGlobalContext<P>) {
        const { context: propsContext, ...others } = props;
        return (
            <ContextLayout>
                {context => (
                    <ContextProvider {...{ ...context, ...propsContext }}>
                        <Component {...(others as P)} />
                    </ContextProvider>
                )}
            </ContextLayout>
        );
    };
}
