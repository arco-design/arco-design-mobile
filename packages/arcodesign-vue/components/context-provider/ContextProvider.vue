<script lang="ts" setup>
import { provide } from 'vue';
import { GlobalContextParams } from './type';
import { configProviderInjectionKey, defaultContext } from './utils';

const props = withDefaults(defineProps<GlobalContextParams>(), defaultContext);
provide(configProviderInjectionKey, props);
</script>
<template>
    <slot />
</template>
