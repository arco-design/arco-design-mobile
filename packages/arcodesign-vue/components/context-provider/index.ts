import ContextProvider from './ContextProvider.vue';
import { installFactory } from '../_helpers/install';

export type * from './type';
export * from './utils';

const install = installFactory(ContextProvider);

export { ContextProvider };

export default {
    install,
};
