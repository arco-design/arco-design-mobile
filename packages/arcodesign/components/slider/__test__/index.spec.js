import React, { useState } from 'react';
import '@testing-library/jest-dom';
import { fireEvent, render } from '@testing-library/react';
import demoTest from '../../../tests/demoTest';
import mountTest from '../../../tests/mountTest';
import Slider from '..';
import { defaultContext } from '../../context-provider';
import {
    createStartTouchEventObject,
    createMoveTouchEventObject,
} from '../../../tests/helpers/mockEvent';

const prefix = `${defaultContext.prefixCls}-slider`;

demoTest('slider');

mountTest(Slider, 'Slider');

function Test(props) {
    const [value, setValue] = useState(20);

    const onChange = newValue => {
        setValue(newValue);
    };

    return <Slider date-testid="slider" value={value} onChange={onChange} {...props} />;
}

function fakeSliderLineRect(component) {
    const line = component.querySelector(`.${prefix}-line`);
    line.getBoundingClientRect = jest.fn(() => ({
        top: 600,
        bottom: 602,
        left: 10,
        right: 110,
        width: 100,
        height: 2,
    }));
}

describe('Slider', () => {
    it('render correctly', async () => {
        const { container } = render(<Test />);

        expect(container.querySelectorAll(`.${prefix}`).length).toBe(1);
        expect(container.querySelectorAll(`.${prefix}-line`).length).toBe(2);
        expect(container.querySelectorAll(`.${prefix}-line.is-activated`).length).toBe(1);
        expect(container.querySelectorAll(`.${prefix}-thumb`).length).toBe(1);
    });

    it('render range slider correctly', async () => {
        const { container } = render(<Slider range defaultValue={[15, 30]} />);
        fakeSliderLineRect(container);
        expect(container.querySelectorAll(`.${prefix}-thumb`).length).toBe(2);
        // 测试在范围区间滑动是否正确
        const line = container.querySelector(`.${prefix}-line`);
        await fireEvent.touchStart(line, createStartTouchEventObject({ x: 30, y: 1 }));
        await fireEvent.touchMove(line, createMoveTouchEventObject({ x: 50, y: 1 }));
        await fireEvent.touchEnd(line, createStartTouchEventObject({ x: 50, y: 1 }));

        expect(container.querySelector(`.${prefix}-popover-content`)).toHaveTextContent('40');
    });

    it('marks slider', () => {
        const { container } = render(<Test max={20} showMarks marks={[0, 5, 10, 15, 20]} />);
        expect(container.querySelector(`.${prefix}`).classList.contains('show-marks')).toBe(true);
        expect(container.querySelectorAll(`.${prefix}-marks-item`).length).toBe(5);
        expect(container.querySelectorAll(`.${prefix}-marks-item-label`).length).toBe(5);
        // component.setState({ value: 10 });
        // expect(component.find(`.${prefix}-marks-item`).find('.is-activated').length).toBe(5);
    });

    it('touch event', async () => {
        const { container } = render(<Test />);
        fakeSliderLineRect(container);

        const line = container.querySelector(`.${prefix}-line`);
        await fireEvent.touchStart(line, createStartTouchEventObject({ x: 20, y: 1 }));
        await fireEvent.touchMove(line, createMoveTouchEventObject({ x: 50, y: 1 }));
        await fireEvent.touchEnd(line, createStartTouchEventObject({ x: 50, y: 1 }));

        expect(container.querySelector(`.${prefix}-line.is-activated`).style.width).toBe('40px');
    });

    it('set the number of segments correctly', async () => {
        const { container } = render(
            <Slider showMarks useMarkOnly max={20} marks={[0, 5, 10, 15, 20]} defaultValue={5} />,
        );

        fakeSliderLineRect(container);
        const line = container.querySelector(`.${prefix}-line`);
        await fireEvent.touchStart(line, createStartTouchEventObject({ x: 20, y: 1 }));
        await fireEvent.touchMove(line, createMoveTouchEventObject({ x: 50, y: 1 }));
        await fireEvent.touchEnd(line, createStartTouchEventObject({ x: 50, y: 1 }));

        expect(container.querySelector(`.${prefix}`).getAttribute('aria-valuenow')).toBe('10');
        expect(container.querySelector(`.${prefix}-line.is-activated`).style.width).toBe('50px');
    });

    it('render tooltip correctly', () => {
        const { container } = render(<Slider showTooltip="always" defaultValue={32} />);
        expect(container.querySelectorAll(`.${prefix}-popover`).length).toBe(1);
    });
});
