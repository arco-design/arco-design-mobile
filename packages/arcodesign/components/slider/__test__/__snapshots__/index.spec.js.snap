// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`slider demo test slider demo: always.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-disabled="false"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="32"
    class="arco-slider  is-horizontal show-tooltip"
    role="slider"
    style="padding-top: 61px;"
  >
    <div
      class="arco-slider-wrapper"
    >
      <div
        class="arco-slider-marks"
      />
      <div
        class="arco-slider-line"
        style="height: 2px;"
      >
        <div
          class="arco-slider-line is-activated is-transferable"
          style="height: 2px; left: 0px; width: 0px;"
        >
          <div
            class="arco-slider-thumb-wrapper is-end"
          >
            <div
              class="arco-slider-popover-wrapper"
            >
              <div
                class="arco-slider-popover"
              >
                <div
                  class="arco-slider-popover-content"
                >
                  32
                </div>
                <div
                  class="arco-slider-popover-arrow"
                />
              </div>
              <div
                class="arco-slider-thumb"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`slider demo test slider demo: custom-btn.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-disabled="false"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="32"
    class="arco-slider  is-horizontal"
    role="slider"
  >
    <div
      class="arco-slider-wrapper"
    >
      <div
        class="arco-slider-marks"
      />
      <div
        class="arco-slider-line"
        style="height: 2px;"
      >
        <div
          class="arco-slider-line is-activated is-transferable"
          style="height: 2px; left: 0px; width: 0px;"
        >
          <div
            class="arco-slider-thumb-wrapper is-end"
          >
            <div
              class="slider-custom-thumb-two"
            >
              32
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`slider demo test slider demo: custom-color.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-disabled="false"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="32"
    class="arco-slider slider-custom-yellow is-horizontal"
    role="slider"
  >
    <div
      class="arco-slider-wrapper"
    >
      <div
        class="arco-slider-marks"
      />
      <div
        class="arco-slider-line"
        style="height: 2px;"
      >
        <div
          class="arco-slider-line is-activated is-transferable"
          style="height: 2px; left: 0px; width: 0px;"
        >
          <div
            class="arco-slider-thumb-wrapper is-end"
          >
            <div
              class="arco-slider-thumb"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`slider demo test slider demo: custom-icon.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-disabled="false"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="27"
    class="arco-slider  is-horizontal"
    role="slider"
  >
    <div
      class="arco-slider-label is-prefix"
    >
      <svg
        class="arco-icon arco-icon-notice-off "
        fill="currentColor"
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M118.101 57.75L962.86 902.506a21.333 21.333 0 010 30.165l-30.187 30.187a21.333 21.333 0 01-30.165 0L57.749 118.1a21.333 21.333 0 010-30.186l30.166-30.166a21.333 21.333 0 0130.186 0zm500.566 795.583A21.333 21.333 0 01640 874.667v42.666a21.333 21.333 0 01-21.333 21.334H405.333A21.333 21.333 0 01384 917.333v-42.666a21.333 21.333 0 0121.333-21.334h213.334zM190.25 341.1l69.888 69.888A280.043 280.043 0 00256 459.136V704h297.152l85.333 85.333H106.667A21.333 21.333 0 0185.333 768v-42.155a21.333 21.333 0 0121.334-21.333h.512l63.488 1.515V459.008c0-41.365 6.912-81.067 19.584-117.91zM576 64a21.333 21.333 0 0121.333 21.333v32.427c147.222 39.125 256 177.067 256 341.248v243.499L768 617.173V459.157C768 311.595 653.376 192 512 192c-48.021 0-92.97 13.803-131.37 37.803l-61.526-61.526a334.763 334.763 0 01107.563-50.517V85.333A21.333 21.333 0 01448 64h128z"
        />
      </svg>
    </div>
    <div
      class="arco-slider-wrapper"
    >
      <div
        class="arco-slider-marks"
      />
      <div
        class="arco-slider-line"
        style="height: 2px;"
      >
        <div
          class="arco-slider-line is-activated is-transferable"
          style="height: 2px; left: 0px; width: 0px;"
        >
          <div
            class="arco-slider-thumb-wrapper is-end"
          >
            <div
              class="arco-slider-thumb"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="arco-slider-label is-suffix"
    >
      <svg
        class="arco-icon arco-icon-notice "
        fill="currentColor"
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M618.667 853.333A21.333 21.333 0 01640 874.667v42.666a21.333 21.333 0 01-21.333 21.334H405.333A21.333 21.333 0 01384 917.333v-42.666a21.333 21.333 0 0121.333-21.334h213.334zM576 64a21.333 21.333 0 0121.333 21.333v32.427c147.222 39.125 256 177.067 256 341.248V704h64a21.333 21.333 0 0121.334 21.333V768a21.333 21.333 0 01-21.334 21.333H106.667A21.333 21.333 0 0185.333 768v-42.667A21.333 21.333 0 01106.667 704h64V459.008c0-164.181 108.8-302.123 256-341.248V85.333A21.333 21.333 0 01448 64h128zm-64 128c-141.376 0-256 119.595-256 267.136V704h512V459.136C768 311.595 653.376 192 512 192z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`slider demo test slider demo: custom-popover.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-disabled="false"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="10"
    class="arco-slider  is-horizontal show-tooltip"
    role="slider"
    style="padding-top: 60px;"
  >
    <div
      class="arco-slider-wrapper"
    >
      <div
        class="arco-slider-marks"
      />
      <div
        class="arco-slider-line"
        style="height: 2px;"
      >
        <div
          class="arco-slider-line is-activated is-transferable"
          style="height: 2px; left: 0px; width: 0px;"
        >
          <div
            class="arco-slider-thumb-wrapper is-start"
          >
            <div
              class="arco-slider-thumb"
            />
          </div>
          <div
            class="arco-slider-thumb-wrapper is-end"
          >
            <div
              class="arco-popover black-theme follow-mode"
            >
              <div
                class="popover-child-inner"
              >
                <div
                  class="arco-slider-thumb"
                />
              </div>
              <div
                class="arco-popover-inner popover-inner all-border-box follow-mode black-theme"
                style="min-width: 10px; max-width: 90vw; transform-origin: 0 0;"
              >
                <div
                  class="popover-bg"
                >
                  <div
                    class="popover-arrow topRight"
                    style="width: 9px; height: 9px;"
                  >
                    <div
                      class="popover-arrow-content"
                    />
                  </div>
                </div>
                <div
                  class="popover-content topRight"
                >
                  <div
                    class="content-text"
                  >
                    value: 60
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`slider demo test slider demo: custom-size.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-disabled="false"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="32"
    class="arco-slider  is-horizontal"
    role="slider"
  >
    <div
      class="arco-slider-wrapper"
    >
      <div
        class="arco-slider-marks"
      />
      <div
        class="arco-slider-line"
        style="height: 4px;"
      >
        <div
          class="arco-slider-line is-activated is-transferable"
          style="height: 4px; left: 0px; width: 0px;"
        >
          <div
            class="arco-slider-thumb-wrapper is-end"
          >
            <div
              class="arco-slider-thumb"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`slider demo test slider demo: disabled.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-disabled="true"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="32"
    class="arco-slider  is-horizontal is-disabled"
    role="slider"
  >
    <div
      class="arco-slider-wrapper"
    >
      <div
        class="arco-slider-marks"
      />
      <div
        class="arco-slider-line"
        style="height: 2px;"
      >
        <div
          class="arco-slider-line is-activated is-transferable"
          style="height: 2px; left: 0px; width: 0px;"
        >
          <div
            class="arco-slider-thumb-wrapper is-end"
          >
            <div
              class="arco-slider-popover-wrapper"
            >
              <div
                class="arco-slider-thumb"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`slider demo test slider demo: index.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-disabled="false"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="32"
    class="arco-slider  is-horizontal"
    role="slider"
  >
    <div
      class="arco-slider-wrapper"
    >
      <div
        class="arco-slider-marks"
      />
      <div
        class="arco-slider-line"
        style="height: 2px;"
      >
        <div
          class="arco-slider-line is-activated is-transferable"
          style="height: 2px; left: 0px; width: 0px;"
        >
          <div
            class="arco-slider-thumb-wrapper is-end"
          >
            <div
              class="arco-slider-thumb"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`slider demo test slider demo: mark.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-disabled="false"
    aria-valuemax="20"
    aria-valuemin="0"
    aria-valuenow="5"
    class="arco-slider  is-horizontal show-tooltip show-marks"
    role="slider"
  >
    <div
      class="arco-slider-wrapper"
    >
      <div
        class="arco-slider-marks"
      >
        <div
          class="arco-slider-marks-item is-activated is-transferable"
          style="left: 0%;"
        >
          <div
            class="arco-slider-marks-item-label"
          >
            <p>
              0
            </p>
          </div>
        </div>
        <div
          class="arco-slider-marks-item is-activated is-transferable"
          style="left: 25%;"
        >
          <div
            class="arco-slider-marks-item-label"
          >
            <p>
              5
            </p>
          </div>
        </div>
        <div
          class="arco-slider-marks-item is-transferable"
          style="left: 50%;"
        >
          <div
            class="arco-slider-marks-item-label"
          >
            <p>
              10
            </p>
          </div>
        </div>
        <div
          class="arco-slider-marks-item is-transferable"
          style="left: 75%;"
        >
          <div
            class="arco-slider-marks-item-label"
          >
            <p>
              15
            </p>
          </div>
        </div>
        <div
          class="arco-slider-marks-item is-transferable"
          style="left: 100%;"
        >
          <div
            class="arco-slider-marks-item-label"
          >
            <p>
              20
            </p>
          </div>
        </div>
      </div>
      <div
        class="arco-slider-line"
        style="height: 2px;"
      >
        <div
          class="arco-slider-line is-activated is-transferable"
          style="height: 2px; left: 0px; width: 0px;"
        >
          <div
            class="arco-slider-thumb-wrapper is-end"
          >
            <div
              class="arco-slider-popover-wrapper"
            >
              <div
                class="arco-slider-thumb"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`slider demo test slider demo: range.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-disabled="false"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="15"
    class="arco-slider  is-horizontal show-tooltip"
    role="slider"
  >
    <div
      class="arco-slider-wrapper"
    >
      <div
        class="arco-slider-marks"
      />
      <div
        class="arco-slider-line"
        style="height: 2px;"
      >
        <div
          class="arco-slider-line is-activated is-transferable"
          style="height: 2px; left: 0px; width: 0px;"
        >
          <div
            class="arco-slider-thumb-wrapper is-start"
          >
            <div
              class="arco-slider-popover-wrapper"
            >
              <div
                class="arco-slider-thumb"
              />
            </div>
          </div>
          <div
            class="arco-slider-thumb-wrapper is-end"
          >
            <div
              class="arco-slider-popover-wrapper"
            >
              <div
                class="arco-slider-thumb"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`slider demo test slider demo: range-tag.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-disabled="false"
    aria-valuemax="10"
    aria-valuemin="0"
    aria-valuenow="1"
    class="arco-slider  is-horizontal show-tooltip"
    role="slider"
    style="padding-top: 41px;"
  >
    <div
      class="arco-slider-label is-prefix"
    >
      0
    </div>
    <div
      class="arco-slider-wrapper"
    >
      <div
        class="arco-slider-marks"
      />
      <div
        class="arco-slider-line"
        style="height: 2px;"
      >
        <div
          class="arco-slider-line is-activated is-transferable"
          style="height: 2px; left: 0px; width: 0px;"
        >
          <div
            class="arco-slider-thumb-wrapper is-start"
          >
            <div
              class="arco-slider-popover-wrapper"
            >
              <div
                class="arco-slider-popover"
              >
                <div
                  class="arco-slider-popover-content"
                >
                  1 thousand
                </div>
                <div
                  class="arco-slider-popover-arrow"
                />
              </div>
              <div
                class="arco-slider-thumb"
              />
            </div>
          </div>
          <div
            class="arco-slider-thumb-wrapper is-end"
          >
            <div
              class="arco-slider-popover-wrapper"
            >
              <div
                class="arco-slider-popover"
              >
                <div
                  class="arco-slider-popover-content"
                >
                  6 thousand
                </div>
                <div
                  class="arco-slider-popover-arrow"
                />
              </div>
              <div
                class="arco-slider-thumb"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="arco-slider-label is-suffix"
    >
      10
    </div>
  </div>
</DocumentFragment>
`;

exports[`slider demo test slider demo: vertical.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-disabled="false"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="40"
    class="arco-slider  is-vertical show-tooltip"
    role="slider"
    style="height: 210px;"
  >
    <div
      class="arco-slider-wrapper"
    >
      <div
        class="arco-slider-marks"
      />
      <div
        class="arco-slider-line"
        style="width: 2px;"
      >
        <div
          class="arco-slider-line is-activated is-transferable"
          style="width: 2px; bottom: 0px; height: 0px;"
        >
          <div
            class="arco-slider-thumb-wrapper is-end"
          >
            <div
              class="arco-slider-popover-wrapper"
            >
              <div
                class="arco-slider-thumb"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
